internal/services/functions/unique_string_function.go

```diff
@@ -0,0 +1,160 @@
+package functions
+
+import (
+	"context"
+	"math/bits"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type UniqueStringFunction struct{}
+
+func (b *UniqueStringFunction) Metadata(ctx context.Context, request function.MetadataRequest, response *function.MetadataResponse) {
+	response.Name = "unique_string"
+}
+
+func (b *UniqueStringFunction) Definition(ctx context.Context, request function.DefinitionRequest, response *function.DefinitionResponse) {
+	response.Definition = function.Definition{
+		Parameters: []function.Parameter{
+			function.ListParameter{
+				ElementType:         types.StringType,
+				AllowNullValue:      false,
+				AllowUnknownValues:  false,
+				Name:                "base_string",
+				Description:         "The values used in the hash function to create a unique string.",
+				MarkdownDescription: "The values used in the hash function to create a unique string.",
+			},
+		},
+		Return:              function.StringReturn{},
+		Summary:             "Creates a deterministic hash string based on the values provided as parameters.",
+		Description:         "This function constructs an Azure equivalent uniqueString value. It is useful for migrating existing resources based on th ARM uniqueString function.",
+		MarkdownDescription: "This function constructs an Azure equivalent `uniqueString` value. It is useful for migrating existing resources based on th ARM `uniqueString` function.",
+	}
+}
+
+func (b *UniqueStringFunction) Run(ctx context.Context, request function.RunRequest, response *function.RunResponse) {
+	var baseString types.List
+
+	if response.Error = request.Arguments.Get(ctx, &baseString); response.Error != nil {
+		return
+	}
+
+	var slice []string
+	if diagnostics := baseString.ElementsAs(ctx, &slice, false); diagnostics.HasError() {
+		response.Error = function.FuncErrorFromDiags(ctx, diagnostics)
+		return
+	}
+
+	uniqueString := uniqueString(slice...)
+
+	response.Error = response.Result.Set(ctx, types.StringValue(uniqueString))
+}
+
+var _ function.Function = &UniqueStringFunction{}
+
+func uniqueString(values ...string) string {
+	value := strings.Join(values, "-")
+	hash := murmurHash64(value)
+	return base32Encode(hash)
+}
+
+func base32Encode(value uint64) string {
+	const text = "abcdefghijklmnopqrstuvwxyz234567"
+	var builder strings.Builder
+	for i := 0; i < 13; i++ {
+		builder.WriteByte(text[int32(value>>59)])
+		value <<= 5
+	}
+	return builder.String()
+}
+
+func murmurHash64(value string) uint64 {
+	bytes := []byte(value)
+	return murmurHash64A(bytes, 0)
+}
+
+func murmurHash64A(data []byte, seed uint32) uint64 {
```

Comment:
Can we add unit tests for this function as https://github.com/smokedlinq/powershell-az/blob/d5461aa62614ee741275b18ba7f710fb3e6b0856/powershell-az.tests.ps1?